src/lib.rs

```diff
@@ -180,8 +197,21 @@ impl<T> StableRegion<T> {
             // We could choose a different rule here if we wanted to be
             // more conservative with memory (e.g. page size allocations).
             let next_len = (self.local.capacity() + 1).next_power_of_two();
-            let new_local = Vec::with_capacity(std::cmp::max(count, next_len));
+            let capacity = std::cmp::max(count, next_len);
+            let capacity = std::cmp::max(capacity, 1<<12 / std::mem::size_of::<T>());
+            let file = tempfile::tempfile().unwrap();
+            let byte_len = std::mem::size_of::<T>() * capacity;
+            let (mmap, new_local) = unsafe {
+                use std::os::unix::io::AsFd;
+                libc::fallocate(mem::transmute::<_, i32>(file.as_fd()), 0, 0, byte_len as i64);
+                let mmap = memmap2::MmapRaw::map_raw(&file).unwrap();
```

Comment:
Why not throw in an `madvise MADV_SEQUENTIAL` here or `fadvise` on the file since we want to experiment with it as a hunch?

Also we should consider advising users of this to tune their `vm.dirty_*` sysctls appropriately.

Reply:
Up for whatever for experimentation, but I think we don't necessarily expect the access patterns to be sequential. I'm not sure what the hint causes to happen, but e.g. 1. non-temporal reads would be fine, but 2. prefetching next pages might not pan out.

Reply:
We should also consider a wrapper around this which does `advise(Advice::DontNeed)` before dropping the mmap.

Reply:
I did some benchmarking and I want to retract the DONTNEED suggestion; it seems Linux always does the wrong thing here, no matter what advice you give it.  I'll follow up on this outside this PR.